catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -107,6 +109,21 @@ public MetacardTypeImpl(String name, MetacardType metacardType,
         descriptors.addAll(additionalDescriptors);
     }
 
+    /**
+     * Creates a {@code MetacardTypeImpl} with the provided {@code name} and
+     * {@link MetacardType}s.
+     *
+     * @param name        the name of this {@code MetacardTypeImpl}
+     * @param metacardTypes the set of {@link MetacardType}s to compose this {@code MetacardTypeImpl}
+     */
+    public MetacardTypeImpl(String name, List<MetacardType> metacardTypes) {
+        this.name = name;
+        descriptors.addAll(new CoreAttributes().getAttributeDescriptors());
+        if (metacardTypes != null) {
+            metacardTypes.forEach(metacardType -> descriptors.addAll(metacardType.getAttributeDescriptors()));
```

Comment:
This allows duplicate AttributeDescriptors where the name is the same but other values are different.  You should track attribute names as they are added and not add later AttributeDescriptors with the same name that may appear later in the list.  This will protect from duplicate named AttributeDescriptors and allow ties to be broken by list order.